src/relay/transforms/type_infer.cc

```diff
@@ -395,7 +400,7 @@ class TypeInferencer : private ExprFunctor<Type(const Expr&)>,
   //
   // The result will be the return type of the operator.
   Type PrimitiveCall(const FuncTypeNode* op, Array<Type> arg_types, const Attrs& attrs,
-                     const Span& span) {
+                     const Span& span, const Expr& expr) {
```

Comment:
I think you don't use expr in this function at all? Also can you change op to func_type_node :-)

Reply:
Good catch -- I needed it to debug a failure at some point but forgot to back it out again.

BTW we very much need spans everywhere to help debugging.